junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestParameterResolver.java

```diff
@@ -64,4 +74,22 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
 				ex);
 		}
 	}
+
+	public static boolean isAggregate(Parameter parameter) {
+		return ArgumentsAccessor.class.isAssignableFrom(parameter.getType())
+				|| AnnotationUtils.isAnnotated(parameter, AggregateWith.class);
+	}
+
+	private Object aggregate(ParameterContext parameterContext, ExtensionContext extensionContext) {
+
+		Parameter parameter = parameterContext.getParameter();
+		Optional<AggregateWith> annotation = AnnotationUtils.findAnnotation(parameter, AggregateWith.class);
+		ArgumentsAccessor accessor = new DefaultArgumentsAccessor(arguments);
+		// @formatter:off
+		return annotation.map(AggregateWith::value)
+				.map(clazz -> ReflectionUtils.newInstance(clazz))
+				.map(aggregator -> aggregator.aggregateArguments(accessor,parameterContext))
```

Comment:
This needs exception handling analogous that found in the `convert()` method.

Reply:
If i understand correctly, ParameterResolution semantically encompasses any operation that results to the final value? Which means that i could move the Exception to the "resolveParameter" method to account for anything that goes wrong in the process?

Reply:
Well, it depends on how much _contextual information_ needs to be provided to the exception when you create it.

If it's generic, it could be at a higher level; however, if meaningful information is only available at a lower level, the exception needs to be created there with the contextual information.

In other words, when we throw an exception we do our best to provide meaningful information to the user to help with diagnosing the problem.